### PR TITLE
Notify Gitcoin/Funder that Developer is Starting Work

### LINKED ISSUE #683: Notify Gitcoin/Funder that Developer is Starting Work
### User Story

As a developer/contributor, I want to notifiy Gitcoin that I'm starting work in one place.

### Description

[comment]: # (Feature or Bug? i.e Type: Bug)
*Type*: Feature

[comment]: # (Describe the problem and why this task is needed. Provide description of the current state, what you would like to happen, and what actually happen)
*Summary*: Currently users make a comment on the Github issue that they'd like to start work, but sometimes forget to go to Gitcoin to click the start work button. Sometimes users also are not aware that they should ask questions about the ticket or provide an approach to the funder.

### Expected Behavior
User should notify Gitcoin in just one place by clicking start work button. A short form opens asking the user a few questions and when user clicks submit, the notes automatically get posted to Github and we accept this as the user has started work.

### Solution
[comment]: # (Provide a summary of the solution and a task list on what needs to be fixed.)
*Summary*: Please see attached wireframes and designs.

### Definition of Done
- All button states are implemented.
- Form and confirmation is implemented.
- GitcoinBot comment is updated to include the commentary that the developer provided directly in the 
   Gitcoin form, should appear on the Github thread (see wireframe).

### Additional Information
Please work with @thelostone-mc who is building #419 (Issue details page)
Please review with @PixelantDesign

### Materials
startwork_1

startwork_spec

[Gitcoin - Start Issue.pdf](https://github.com/gitcoinco/web/files/1835873/Gitcoin.-.Start.Issue.pdf)

### PR DESCRIPTION
**Pull request by darkdarkdragon**
Add dialog to provide question when starting work.

- [x] linter status: 100% pass
- [x] changes don't break existing behavior
- [x] commit message follows [commit guidelines](https://github.com/gitcoinco/web/blob/master/docs/CONTRIBUTING.md#step-4-commit)

##### Affected core subsystem(s)
ui

##### Testing
manually tested

##### Refers/Fixes
Fixes #683

### CODE REVIEW

**Comment by darkdarkdragon** [comment]
@owocki can't submit work, says "No active bounty found for this Github URL."

File: app/assets/v2/css/base.css [hunk 1]
```diff
@@ -286,6 +286,9 @@ p a {
 .modal a.close-modal {
   display: none !important;
 }
+.modal-close {
```
Comment (thelostone-mc) [inline app/assets/v2/css/base.css]: mind adding an empty line ?

File: app/assets/v2/css/dashboard.css [hunk 2]
```diff
@@ -118,3 +118,4 @@ body {
     font-size: 10px;
   }
 }
+
```
Comment (thelostone-mc) [inline app/assets/v2/css/dashboard.css]: Could we remove the empty line ?

File: app/assets/v2/css/bounty.css [hunk 3]
```diff
@@ -435,3 +435,38 @@ a.btn {
...
+  flex-direction: row;
+  justify-content: flex-start;
+}
+
+.add-interest-options > .option {
+  width:20%;
+}
+.btn-interested {
+  border-radius:0.2em;
+}
```
Comment (thelostone-mc) [inline app/assets/v2/css/bounty.css]: Could we keep it uniform and add a line between css classes?

**Comment by thelostone-mc** [comment]
@darkdarkdragon Ah that's cause the issue got expired :/ 
owocki would be able to help you out with that 

PS: could you run `npm run stylelint` & `npm run eslint` to ensure our linters pass ?

**Comment by thelostone-mc** [comment]
@darkdarkdragon Could you give the button a bit of breathing space (`margin-top` maybe?  )
Also the dialog box doesn't seem to render right on mobile (portrait) and it seems to be overflowing (making it scrollable) on landscape view

x

File: app/assets/v2/js/pages/bounty_details.js [hunk 4]
```diff
@@ -387,17 +387,44 @@ var attach_work_actions = function() {
   $('body').delegate('a[href="/interested"], a[href="/uninterested"]', 'click', function(e) {
     e.preventDefault();
     if ($(this).attr('href') == '/interested') {
-      $(this).attr('href', '/uninterested');
-      $(this).find('span').text('Stop Work');
-      add_interest(document.result['pk']);
-    } else {
+      show_interest_modal.call(this);
+    } else if (confirm('Are you sure you want to remove your interest?')) {
```
Comment (mbeacom) [inline app/assets/v2/js/pages/bounty_details.js]: We'll want to wrap this copy in `gettext()` for i18n support.

File: app/assets/v2/js/pages/bounty_details.js [hunk 5]
```diff
@@ -387,17 +387,44 @@ var attach_work_actions = function() {
...
+      });
+
+      modal.on('submit', function(event) {
+        event.preventDefault();
+
+        var has_question = event.target[0].value;
+        var issue_message = event.target[2].value;
+
+        $(self).attr('href', '/uninterested');
+        $(self).find('span').text('Stop Work');
```
Comment (mbeacom) [inline app/assets/v2/js/pages/bounty_details.js]: We'll want to wrap this copy in `gettext()` for i18n support.

File: app/dashboard/models.py [hunk 6]
```diff
@@ -816,7 +816,9 @@ class Interest(models.Model):
 
     profile = models.ForeignKey('dashboard.Profile', related_name='interested', on_delete=models.CASCADE)
     created = models.DateTimeField(auto_now_add=True, blank=True, null=True)
-
+    has_question = models.BooleanField(default=False)
+    issue_message = models.TextField(default='', blank=True)
+
```
[the remark this answers is not in the document]
Comment (mbeacom) [reply app/dashboard/models.py]: What he said.  Should be a matter of just trimming the whitespace on the new line 821

File: app/dashboard/templates/addinterest.html [hunk 7]
```diff
@@ -0,0 +1,55 @@
+{% comment %}
+    Copyright (C) 2018 Gitcoin Core
```
Comment (mbeacom) [inline app/dashboard/templates/addinterest.html]: Can you bring the indent on this in one for the whole comment block?

File: app/dashboard/templates/addinterest.html [hunk 8]
```diff
@@ -0,0 +1,55 @@
...
+{% load i18n static %}
+<div id="add_interest" class="white-light-bg">
+  <div class="container-fluid">
+    <div class="row closebtn">
+      <a rel="modal:close" href="javascript:void" class="close" aria-label="Close dialog">
+        <span aria-hidden="true">&times;</span>
+      </a>
+    </div>
+    <div class="row">
+      <h5 class="interest-modal-title">Great to see that you're interested in this project!</h5>
```
Comment (mbeacom) [inline app/dashboard/templates/addinterest.html]: All instances of copy throughout this file should be wrapped in `{% trans "<str>" %}`

For example:

`{% trans "Great to see that you're interested in this project!" %}`

File: app/dashboard/views.py [hunk 9]
```diff
@@ -125,6 +124,14 @@ def gh_login(request):
...
 
 
+def get_interest_modal(request):
+    context = {
+        'active': 'get_interest_modal',
+        'title': 'Add Interest',
+    }
+    return TemplateResponse(request, 'addinterest.html', context)
+
+@require_POST
```
[the remark this answers is not in the document]
Comment (mbeacom) [reply app/dashboard/views.py]: We'll want another new line between `return TemplateResponse(request, 'addinterest.html', context)` and `@require_POST` here

**Comment by owocki** [comment]
> PS: could you run npm run stylelint & npm run eslint to ensure our linters pass ?

`make fix` will work too

**Comment by darkdarkdragon** [comment]
@thelostone-mc fixed

**Comment by darkdarkdragon** [comment]
@owocki fixed
just on thing - I've made `work_started` github comment to look like in pdf:
screen shot 2018-04-25 at 22 53 18

but I'm not sure what this is what you want now

**Comment by thelostone-mc** [comment]
All other devices look alright except mobile!

screen shot 2018-04-26 at 2 00 29 pm

@PixelantDesign could you check the UI and see if changes are alright ? (you can checkout the gif I've attached in the comments above)

**Comment by owocki** [comment]
> just on thing - I've made work_started github comment to look like in pdf:

@darkdarkdragon this looks really good!

**Comment by darkdarkdragon** [comment]
@thelostone-mc 
> All other devices look alright except mobile!

fixed, screenshot in next comment

> I'm not able to select No, has it been disabled for some reason ?

it is always enabled, not sure what went wrong
> Ideally we hide the text area and we don't fire an API to github if the user selects No

from previous discussion by @PixelantDesign :

> The radio button is not meant to be progressive. A user could not have any questions but also have a concern/info/ or approach they want to share with the funder/project owner.

**Comment by darkdarkdragon** [comment]
screen shot 2018-04-26 at 19 03 22

**Comment by owocki** [comment]
> @owocki can't submit work, says "No active bounty found for this Github URL."

i'll just pay you via a tip and set the work as 'done' in admin.. sorry about this.  its a bug that we recently fixed but can't make the fix apply to bounties already on the smart contract!

**Comment by owocki** [comment]
testing now

File: app/dashboard/models.py [hunk 10]
```diff
@@ -816,6 +816,8 @@ class Interest(models.Model):
 
     profile = models.ForeignKey('dashboard.Profile', related_name='interested', on_delete=models.CASCADE)
     created = models.DateTimeField(auto_now_add=True, blank=True, null=True)
+    has_question = models.BooleanField(default=False)
+    issue_message = models.TextField(default='', blank=True)
```
Comment (owocki) [inline app/dashboard/models.py]: you need to create a migraiton for these model fields via `./manage.py makemigraitons`

Comment (owocki) [reply app/dashboard/models.py]: and commit it back up

**Comment by owocki** [comment]
```
> @user has committed to working on this project to be completed in 364 days.
```

RIght now this is the copy on the gitcoinbot comment.  @vs77bb what do you think about modifying it to make the user agree that they must start within 24 hours?  and if not we can auto kick the user

File: app/dashboard/models.py (hunk 10 above)
Comment (owocki) [reply app/dashboard/models.py]: 32f56bf8ecc4c3dc5bb83dee40becb9d5ad40511 adds these migrations

File: app/dashboard/notifications.py [hunk 11]
```diff
@@ -331,17 +332,25 @@ def build_github_notification(bounty, event_name, profile_pairs=None):
...
+        msg = f"__Work has been started__.\n{profiles} has committed to working on this project to be " \
+              f"completed in {num} {unit}.\n\n"
+        bounty_owner_clear = f"@{bounty.bounty_owner_github_username}" if bounty.bounty_owner_github_username else ""
+        try:
+            if profile_pairs:
+                for profile in profile_pairs:
+                    interests = Interest.objects.filter(profile__handle=profile[0], bounty=bounty)
+                    for interest in interests:
+                        if interest.issue_message.strip():
+                            msg += f"\n__Please answer following questions/comments__ {bounty_owner_clear}:\n\n" + \
```
[the remark this answers is not in the document]
Comment (darkdarkdragon) [reply app/dashboard/notifications.py]: question is in `Interest` object, so every one will have they own question, if any

File: app/assets/v2/js/pages/bounty_details.js [hunk 12]
```diff
@@ -397,17 +397,44 @@ var attach_work_actions = function() {
   $('body').delegate('a[href="/interested"], a[href="/uninterested"]', 'click', function(e) {
     e.preventDefault();
     if ($(this).attr('href') == '/interested') {
-      $(this).attr('href', '/uninterested');
-      $(this).find('span').text('Stop Work');
-      add_interest(document.result['pk']);
-    } else {
+      show_interest_modal.call(this);
+    } else if (confirm(gettext('Are you sure you want to remove your interest?'))) {
```
Comment (owocki) [inline app/assets/v2/js/pages/bounty_details.js]: +1

File: app/dashboard/notifications.py [hunk 13]
```diff
@@ -331,17 +332,25 @@ def build_github_notification(bounty, event_name, profile_pairs=None):
...
-
-        msg = f"{status_header}__Work has been started on the {natural_value} {bounty.token_name} {usdt_value} funding " \
-              f"by__: \n 1. {profiles} {sub_msg} \n\n * Learn more [on the Gitcoin Issue Details page]({absolute_url})\n " \
-              "* Questions? Checkout <a href='https://gitcoin.co/help'>Gitcoin Help</a> or the <a href='https://gitcoin.co/slack'>Gitcoin Slack</a>\n * " \
-              f"${amount_open_work} more funded OSS Work available on the [Gitcoin Issue Explorer](https://gitcoin.co/explorer)\n"
+        unit = 'day'
+        num = int(round((bounty.expires_date - timezone.now()).days, 0))
+        if num == 0:
+            unit = 'hour'
+            num = int(round((bounty.expires_date - timezone.now()).seconds / 3600 / 24, 0))
```
[the remark this answers is not in the document]
Comment (owocki) [reply app/dashboard/notifications.py]: better yet i wonder if you could use https://docs.djangoproject.com/en/2.0/ref/contrib/humanize/#naturaltime

[the remark this answers is not in the document]
Comment (darkdarkdragon) [reply app/dashboard/notifications.py]: It was copy-paste from other place.
You're right, days would be enough here

**Comment by darkdarkdragon** [comment]
@owocki changed to use `naturaltime`
now comment looks like this
```@vikiniky has committed to working on this project to be completed 1 week, 4 days from now.```

File: app/dashboard/views.py [hunk 14]
```diff
@@ -125,6 +124,15 @@ def gh_login(request):
     return redirect('social:begin', backend='github')
 
 
+def get_interest_modal(request):
+    context = {
+        'active': 'get_interest_modal',
+        'title': 'Add Interest',
```
Comment (mbeacom) [inline app/dashboard/views.py]: Can you wrap this in gettext_lazy? `_()` ?

**Comment by darkdarkdragon** [comment]
@mbeacom @owocki  any news on that one?

**Comment by owocki** [comment]
> @mbeacom @owocki any news on that one?

sorry, what?

**Comment by darkdarkdragon** [comment]
@owocki will it be merged or there is any more feedback?

**Comment by thelostone-mc** [comment]
@darkdarkdragon could we squash the commits into one?

**Comment by darkdarkdragon** [comment]
@thelostone-mc done

File: app/dashboard/migrations/0062_auto_20180426_2227.py [hunk 15]
```diff
@@ -0,0 +1,23 @@
+# Generated by Django 2.0.4 on 2018-04-26 22:27
```
Comment (mbeacom) [inline app/dashboard/migrations/0062_auto_20180426_2227.py]: This migration will need rebuilt as `0063_` before we push this live.

File: app/faucet/migrations/0007_auto_20180426_2227.py [hunk 16]
```diff
@@ -0,0 +1,18 @@
+# Generated by Django 2.0.4 on 2018-04-26 22:27
```
Comment (mbeacom) [inline app/faucet/migrations/0007_auto_20180426_2227.py]: This is present in `master`

**Pull request merged**